cmd/test.go

```diff
@@ -0,0 +1,182 @@
+package cmd
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+	"os"
+	"path"
+	"time"
+
+	"github.com/charmbracelet/log"
+	"github.com/omnigres/cli/orb"
+	"github.com/spf13/cobra"
+)
+
+var testCmd = &cobra.Command{
+	Use:   "test",
+	Short: "Test orbs",
+	Run: func(cmd *cobra.Command, args []string) {
+		var cluster orb.OrbCluster
+		var err error
+		cluster, err = getOrbCluster()
+		if err != nil {
+			log.Fatal(err)
+		}
+
+		log.Debug("Workspace", "workspace", workspace)
+		cwd, err := os.Getwd()
+		if err != nil {
+			log.Fatal(err)
+		}
+
+		orbs, err := currentOrbs(cluster, cwd)
+		if err != nil {
+			log.Fatal(err)
+		}
+
+		ctx := context.Background()
+		log.Debug("Assembling orbs src", "orbs", orbs)
+
+		t := time.Now().Format("20060102150405")
+		nameForTestDatabase := func(orbName string) string {
+			return fmt.Sprintf("%s_%s_%s", orbName, "test", t)
+		}
+
+		err = assembleOrbs(ctx, cluster, true, orbs, nameForTestDatabase)
+		err = testOrbs(ctx, cluster, orbs, nameForTestDatabase)
+
+		if err != nil {
+			log.Fatal(err)
+		}
+	},
+}
+
+func testOrbs(
+	ctx context.Context,
+	cluster orb.OrbCluster,
+	orbs []string,
+	databaseForOrb func(string) string,
+) (err error) {
+
+	var testTarget, testRunner *sql.DB
+	testRunner, err = cluster.Connect(ctx, "omnigres")
+	if err != nil {
+		return
+	}
+
+	testOrb := func(orbName string) error {
+		dbName := databaseForOrb(orbName)
+		testTarget, err = cluster.Connect(ctx, dbName)
+		if err != nil {
+			return err
+		}
+		log.Debug("Testing orb", "orbName", orbName, "dbName", dbName)
+
+		_, err = testRunner.ExecContext(
+			ctx,
+			"update pg_database set datistemplate = true where datname = $1",
+			dbName,
+		)
+		if err != nil {
+			return err
+		}
+		removeIsTemplate := func() {
+			// remove istemplate so we can drop the database
+			_, err = testRunner.ExecContext(
+				ctx,
+				"update pg_database set datistemplate = false where datname = $1",
+				dbName,
+			)
+			if err != nil {
+				log.Fatal(err)
+			}
+		}
+		defer removeIsTemplate()
+
+		_, err = testTarget.ExecContext(ctx, "create extension omni_test cascade")
+		if err != nil {
+			return err
+		}
+		testTarget.Close()
+
+		conn, err := testRunner.Conn(ctx)
+		if err != nil {
+			return err
+		}
+		defer conn.Close()
+
+		// assemble tests in target db
+		orbSource := path.Join(orbName, "test")
+		assemblyRows, err := conn.QueryContext(
```

Comment:
We assemble here using SQL, but what if we generalized an assemble function in CLI to just take "source directory" and database connection and do it? That function handles notification to keep the progress visible.

This way, we can call it twice: assemble(src), assemble(test)

What do you think?